index.js

```diff
@@ -384,155 +272,146 @@ export default class OneSignal {
-        if (!checkIfInitialized()) return;
+        if (!checkIfInitialized(RNOneSignal)) return;
 
         if (Platform.OS === 'android') {
             RNOneSignal.clearOneSignalNotifications();
         } else {
-            console.log("This function is not supported on iOS");
+            console.log("clearOneSignalNotifications: this function is not supported on iOS");
         }
     }
 
     static cancelNotification(id) {
-        if (!checkIfInitialized()) return;
+        if (!checkIfInitialized(RNOneSignal)) return;
 
         if (Platform.OS === 'android') {
             RNOneSignal.cancelNotification(id);
         } else {
-            console.log("This function is not supported on iOS");
+            console.log("cancelNotification: this function is not supported on iOS");
         }
     }
 
-    //Sends MD5 and SHA1 hashes of the user's email address (https://documentation.onesignal.com/docs/ios-sdk-api#section-synchashedemail)
-    static syncHashedEmail(email) {
-        if (!checkIfInitialized()) return;
+    /* E X T E R N A L  U S E R  I D */
 
-        RNOneSignal.syncHashedEmail(email);
-    }
+    static setExternalUserId(externalId, handler) {
+        if (!checkIfInitialized(RNOneSignal)) return;
 
-    static setLogLevel(nsLogLevel, visualLogLevel) {
-        if (!checkIfInitialized()) return;
+        if (handler === undefined)
+            handler = function(){};
 
-        RNOneSignal.setLogLevel(nsLogLevel, visualLogLevel);
+        RNOneSignal.setExternalUserId(externalId, handler);
     }
 
-    static setRequiresUserPrivacyConsent(required) {
-        if (!checkIfInitialized()) return;
-
-        RNOneSignal.setRequiresUserPrivacyConsent(required);
-    }
+    static removeExternalUserId(handler) {
+        if (!checkIfInitialized(RNOneSignal)) return;
 
-    static provideUserConsent(granted) {
-        if (!checkIfInitialized()) return;
+        if (handler === undefined)
+            handler = function(){};
 
-        RNOneSignal.provideUserConsent(granted);
+        RNOneSignal.removeExternalUserId(handler);
     }
 
-    static userProvidedPrivacyConsent() {
-        if (!checkIfInitialized()) return;
+    /* I N  A P P  M E S S A G I N G */
 
-        //returns a promise
-        return RNOneSignal.userProvidedPrivacyConsent();
-    }
-
-    static setExternalUserId(externalId, callback) {
-        if (!checkIfInitialized()) return;
+    static setInAppMessageClickHandler(handler) {
+        if (!checkIfInitialized(RNOneSignal)) return;
+        isValidCallback(handler);
 
-        if (callback === undefined)
-            callback = function(){};
-
-        RNOneSignal.setExternalUserId(externalId, callback);
-    }
-
-    static removeExternalUserId(callback) {
-        if (!checkIfInitialized()) return;
-
-        if (callback === undefined)
-            callback = function(){};
+        if (Platform.OS === 'android') {
+            RNOneSignal.initInAppMessageClickHandlerParams();
```

Comment:
Can we do everything in the Java wrapper when `setInAppMessageClickHandler` is called so we don't need this android check?

Reply:
I assume you mean the ObjC wrapper? We could add an empty stub for it

Reply:
Investigate in the future what `initInAppMessageClickHandlerParams` is used for